pkg/commands/process/settings/rules.go

```diff
@@ -176,27 +176,52 @@ func buildRules(definitions map[string]RuleDefinition, enabledRules map[string]s
 			ruleType = defaultRuleType
 		}
 
+		// build rule trigger
+		ruleTrigger := RuleTrigger{
+			MatchOn:           PRESENCE,
+			DataTypesRequired: false,
+		}
+		if definition.Trigger != nil {
+			if definition.Trigger.MatchOn != nil {
+				ruleTrigger.MatchOn = *definition.Trigger.MatchOn
+			}
+			if definition.Trigger.DataTypesRequired != nil {
+				ruleTrigger.DataTypesRequired = *definition.Trigger.DataTypesRequired
+			}
+			if definition.Trigger.RequiredDetection != nil {
+				ruleTrigger.RequiredDetection = definition.Trigger.RequiredDetection
+			}
+		}
+
+		isLocal := false
+		for _, rulePattern := range definition.Patterns {
+			if strings.Contains(rulePattern.Pattern, "$<DATA_TYPE>") {
+				isLocal = true
+				break
+			}
+		}
```

Comment:
Not sure about this. I need to know, before the rule is run, whether or not it counts as a "local" rule. What I have here feels a bit hack-ish and maybe points to the wider issue that we no longer support rules who have both patterns with `$<DATA_TYPE>` and patterns without `$<DATA_TYPE>` as well as before 🤔

Reply:
definitely feels like the rules should be separated but that prevents us from having a level set to warning (when no data types are detected) and that would increase to critical if we detect something (e.g. gender).
WDYT? @gmontard

Reply:
@cfabianski I've removed the rule split and updated the Rego. Looking forward to your thoughts 🙏